scipy/stats/_stats_py.py

```diff
@@ -401,6 +401,117 @@ def hmean(a, axis=0, dtype=None, *, weights=None):
                          "than or equal to zero")
 
 
+@_axis_nan_policy_factory(
+        lambda x: x, n_samples=1, n_outputs=1, too_small=0, paired=True,
+        result_to_tuple=lambda x: (x,), kwd_samples=['weights'])
```

Comment:
I haven't looked at this in detail, but it looks like you've really done your homework when preparing this PR. Thank you! One thing is to add `pmean` to the test cases in `test_axis_nan_policy`.

Reply:
I have completed `test_axis_nan_policy` with `pmean`, but it generates an error, because `pmean` is not in `stats.mstats`.

[weighted_fun_ma = getattr(stats.mstats, weighted_fun_name)
AttributeError: module 'scipy.stats.mstats' has no attribute 'pmean'](https://github.com/scipy/scipy/runs/5502769569?check_suite_focus=true#step:12:566)

Do I add `pmean` into `stats.mstats`? 
I had understood that it was no longer necessary to add anything in `stats.mstats` because it was just for backward compatibility.
https://github.com/scipy/scipy/pull/14817#discussion_r736350423

Reply:
I actually meant to add it to
https://github.com/scipy/scipy/blob/30d4d7b1b9fd8f081324e6a5b5c2555b413e796d/scipy/stats/tests/test_axis_nan_policy.py#L18-L33

But it also seems useful to add it to `test_gmean_mixed_mask_nan_weights` as you have. You could just skip the part of that test that relies on `mstats`, e.g.:
```python3
if weighted_fun_name not in {'pmean'}:
    weighted_fun_ma = getattr(stats.mstats, weighted_fun_name)
    res5 = weighted_fun_ma(...)
```